Rewrite symbol management system (#543)
ethteck/splat: docs/Adding-Symbols.md

### `segment`

Allows specifying to which specific segment this symbol belongs to, useful to disambiguate symbols from segments that share the same VRAM address.

This name must be the same as the name of a segment listed in the yaml.
The segment's Vram and Rom address must include the addresses of the given symbol.

It is mandatory to either specify at least this or [rom](#rom) for symbols in overlays where multiple symbols could share the same VRAM address. If both are missing, the symbol is assumed to be part of any of the non-overlay segments.

**Example:**
```ini
sMenuTexture = 0x06004040; // segment:menu_assets
```

### `absolute`

Declares this symbol as being associated to no segment.

> Some symbols are special in the way they exist outside the address space of the ROM. Instead they are provided by the OS, hardware, etc. This way you can name these kind of symbols.

Reviewer: ```suggestion
Some symbols are special in the way they exist outside the address space of the ROM. Instead they are provided by the OS, hardware, etc. `absolute` can be used to name these kind of symbols.
```
